src/main/scala/is/hail/sparkextras/OrderedRDD.scala

```diff
@@ -391,7 +392,7 @@ class OrderedRDD[PK, K, V] private(rdd: RDD[(K, V)], val orderedPartitioner: Ord
     (implicit ord: Ordering[(K, V)] = null): RDD[(K, V)] = {
     require(maxPartitions > 0, "cannot coalesce to nPartitions <= 0")
     val n = rdd.partitions.length
-    if (maxPartitions == n || n == 0 || !shuffle && (maxPartitions > n))
+    if (!shuffle && maxPartitions >= n)
```

Comment:
Yes. If shuffle == true, then we want to shuffle the dataset. This is used to rebalance the partitions, even if we end up with the same number as before